### PR TITLE
Check links on CI

### LINKED ISSUE #1745: Dead links
Using pytest-check-links:

```
C:\dev\jupyterlab [master ≡]> py.test --check-links -k .md .
============================= test session starts =============================
platform win32 -- Python 3.5.2, pytest-3.0.6, py-1.4.32, pluggy-0.4.0
rootdir: C:\dev\jupyterlab, inifile: pytest.ini
plugins: tornado-0.4.5, timeout-1.2.0, cov-2.4.0, nbval-0.4.1, check-links-0.0.3.dev4
collected 121 items

CONTRIBUTING.md .......F..
README.md ......................
RELEASE.md ..
design\about.md .
tutorial\adding_content.md ...F
tutorial\documents.md F.........
tutorial\examples.md ..
tutorial\labextensions.md F.
tutorial\notebook.md ..................
tutorial\patterns.md ..
tutorial\plugins.md ...............
tutorial\repo.md .
tutorial\virtualdom.md ..

================================== FAILURES ===================================
___ C:\dev\jupyterlab\CONTRIBUTING.md: http://jupyter.github.io/jupyterlab/ ___
http://jupyter.github.io/jupyterlab/: HTTP Error 404: Not Found
________ C:\dev\jupyterlab\tutorial\adding_content.md: documents.html _________
documents.html: No such file: C:\dev\jupyterlab\tutorial\documents.html
____________ C:\dev\jupyterlab\tutorial\documents.md: plugins.html ____________
plugins.html: No such file: C:\dev\jupyterlab\tutorial\plugins.html
__________ C:\dev\jupyterlab\tutorial\labextensions.md: plugins.html __________
plugins.html: No such file: C:\dev\jupyterlab\tutorial\plugins.html
============================= 30 tests deselected =============================
============= 4 failed, 87 passed, 30 deselected in 40.90 seconds =============
```
The last three are probably ok(?), but the first one is properly broken.

### PR DESCRIPTION
**Pull request by blink1073**
Fixes #1745.  Also fixes intra-repo document links on Github.

### CODE REVIEW

**Comment by blink1073** [comment]
@vidartf, @minrk, big internet hugs for you.

**Comment by blink1073** [comment]
cc @willingc

**Comment by vidartf** [comment]
There might be a better way (more robust) to filter for only `.md` files than `-k .md` (which is basically a regex).

**Comment by vidartf** [comment]
Since this is for Travis, you can probably use shell glob (`py.test --check-link **/*.md`)

**Comment by blink1073** [comment]
Done, thanks!

File: scripts/travis_script.sh [hunk 1]
```diff
@@ -38,6 +38,10 @@ if [[ $GROUP == coverage_and_docs ]]; then
     # Run the coverage check.
     npm run test:coverage
 
+    # Run the link check
+    pip install -q pytest-check-links
+    py.test --check-links **/*.md
```
Comment (vidartf) [inline scripts/travis_script.sh]: Oh, sorry. It should be `py.test --check-link *.md **/*.md`, as `**` seems to map 1-or-more subdir on bash, not 0-or-more.

Comment (vidartf) [inline scripts/travis_script.sh]: Hmm. Testing further locally, this will only match one level down, not N-levels...

Comment (blink1073) [inline scripts/travis_script.sh]: Well that's a bummer.

**Pull request merged**